slot.go

```diff
@@ -169,7 +254,17 @@ var (
 	oidNamedCurveP521 = asn1.ObjectIdentifier{1, 3, 132, 0, 35}
 )
 
-func publicKey(obj obj, t string) (crypto.PublicKey, error) {
+func publicKey(obj obj, c, t string) (crypto.PublicKey, error) {
+	if c == "certificate" {
+		attrs, err := obj.ctx.GetAttributeValue(obj.h, obj.o, []*pkcs11.Attribute{
+			&pkcs11.Attribute{Type: pkcs11.CKA_VALUE},
+		})
+		if err != nil {
+			return nil, err
+		}
+		cert, err := x509.ParseCertificate(attrs[0].Value)
```

Comment:
Fixed the error handling here and added a bit more context to both of the error returns from `slot.publicKey` on the certificate portion.